src/components/MapPanel.jsx

```diff
@@ -1,66 +1,78 @@
-                        className="form-check"
-                        style={{paddingBottom: '1.25rem'}}>
-                        <input
-                            className="form-check-input"
-                            type="checkbox"
-                            value=""
-                            id="flexCheckDefault"
-                            checked={isConnectionMode}
-                            onClick={e => {
-                                console.log(e.target.checked);
-                                connectionModeToggle(e.target.checked);
-                            }}
-                        />
-                        <label
-                            className="form-check-label"
-                            htmlFor="flexCheckDefault">
-                            Connect stops?
-                        </label>
-                    </div>
-                    {isConnectionMode && <ConnectionSidePanel />}
+  const {
+    showSingleTile,
+    isViewMode,
+    isReportMapMode,
+    isConnectionMode,
+    singleTileToggle,
+    viewModeToggle,
+    reportModeToggle,
+    connectionModeToggle,
+    flush,
+  } = useContext(MapContext);
+
+  const radios = [
+    {
+      title: 'See bus stop details',
+      isChecked: isViewMode,
+      onClickHandler: e => {
+        viewModeToggle(e.target.checked);
+      },
+    },
+    {
+      title: 'Create report on map',
+      isChecked: isReportMapMode,
+      onClickHandler: e => {
+        reportModeToggle(e.target.checked);
+      },
+    },
+    {
+      title: 'Create new connection',
+      isChecked: isConnectionMode,
+      onClickHandler: e => {
+        connectionModeToggle(e.target.checked);
+      },
+    },
+  ];
 
-                    <div
-                        className="form-check"
-                        style={{paddingBottom: '1.25rem'}}>
-                        <input
-                            className="form-check-input"
-                            type="checkbox"
-                            value=""
-                            checked="checked"
-                            id="flexCheckDefault"
-                            onClick={e => {
-                                singleTileToggle(e.target.checked);
-                                e.target.checked && flush();
-                            }}
-                        />
-                        <label
-                            className="form-check-label"
-                            htmlFor="flexCheckDefault">
-                            Show single tile
-                        </label>
-                    </div>
-                </Fragment>
-            )}
-        </Fragment>
-    );
+  return (
+    <>
+      {showSingleTile && (
+        <div className="map-panel__container">
+          <div className="map-panel__button">
+            <CustomBlockButton
+              buttonTitle={'Hide tile'}
+              style={buttonStyle}
+              handleOnClick={() => {
```

Comment:
```suggestion
              onClickHandler={() => {
```

Reply:
I don't know the convention in the project, I just like calling properties with nouns